installation.md

```diff
@@ -123,11 +127,43 @@ YouCompleteMe supports many of clangd's features:
 
 - **Alternate clangd binary**: set `g:ycm_clangd_binary_path` in `.vimrc`.
 
----
+### coc-clangd
+
+Supports both Vim and Neovim. [coc-clangd](https://github.com/clangd/coc-clangd) is an extension for
+[coc.nvim](https://github.com/neoclide/coc.nvim), you need to [install coc.nvim](https://github.com/neoclide/coc.nvim/wiki/Install-coc.nvim) first.
+
+`:CocInstall coc-clangd` in Vim/Neovim to install coc-clangd. coc-clangd will try to find clangd from your `$PATH`,
+if not found, run `:CocCommand clangd.install` to install the [latest release](https://github.com/clangd/clangd/releases/latest)
+from GitHub, or set `clangd.path` in `:CocConfig` to use special clangd binary.
+
+coc-clangd provides configurations for clangd, you can set them in `:CocConfig`:
+
+```json
+{
+  "clangd.path": "/path/to/custom/clangd",
+  "clangd.arguments": ["--background-index", "--clang-tidy"],
+  "clangd.fallbackFlags": ["-std=c++23"]
+}
+```
+
+Check [coc-clangd's README](https://github.com/clangd/coc-clangd?tab=readme-ov-file#configurations) for more options.
+
+coc-clangd also provides commands to interact with clangd:
+
+- `:CocCommand clangd.switchSourceHeader`, switch between source/header files
+- `:CocCommand clangd.symbolInfo`, resolve symbol info under the cursor
+- `:CocCommand clangd.memoryUsage`, show memory usage
+- `:CocCommand clangd.install`, install latest release from GitHub
+- `:CocCommand clangd.update`, check and update clangd from GitHub
+
+### Neovim built-in LSP client
+
+Neovim only. Neovim has a built-in LSP client, which can be configured to work with clangd.
+
+1. install [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) with your plugin manager
+2. enable clangd in your init.lua: `require'lspconfig'.clangd.setup{}`
```

Comment:
It would be great to have some more details about how to add arguments, and how to get logs, but this can be left for a future PR